### DISCUSSION TITLE
Support for CH569 - missing patch for interrupt attribute, at least for Raspberry Pi 5 (64 bit)

### DISCUSSION THREAD

**Discussion by capiman** [xpack-dev-tools/riscv-none-elf-gcc-xpack#27] [General]
Hello,

I am only a user of the pre-compiled gcc for RISC-V. I tried to compile source code for the CH569,
( https://www.wch-ic.com/products/CH569.html )
but got problems, because it looks like a CH569 specific interrupt keyword/attribute is not available in the main release.

https://github.com/hydrausb3/wch-ch56x-bsp/blob/eb6318d0c61c05baa527f59c47e5b967e4e9b00e/usb/usb_devbulk/CH56x_usb20_devbulk.c#L546

Without this keyword, the USB interrupt is not running correctly and therefore general USB functionality does not work.

There seems to be already a patch

https://github.com/hydrausb3/riscv-none-elf-gcc-xpack/commit/90893d7a13fec53252313294b12882a5f458c019

but I don't know where to start to bring this patch into a release.
Or at least how I can try out to compile a local release with this patch by my own (for Raspberry Pi 5 64 bit).

Is there some documentation or tutorial which could help me on this task?

Best regards,
Martin

**Comment by capiman** [comment]
Just saw there is the same request here

https://github.com/xpack-dev-tools/riscv-none-elf-gcc-xpack/discussions/7

But 1 year later the patch was not yet integrated. Could there be a new start to integrate this patch into main release?

**Reply by ilg-ul** [reply]
Sorry, but this distribution follows the official GNU GCC project.

When that patch will be integrated upstream, it'll become available here too.
